org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLLinkedEditing.java

```diff
@@ -34,6 +34,12 @@
 
 	private static Logger LOGGER = Logger.getLogger(XMLLinkedEditing.class.getName());
 
+	public static final String XML_ELEMENT_WORD_PART_PATTERN =

```

Comment:
I'm not an expert inregexp, but are you sure that it's the same regexp than https://github.com/microsoft/vscode/blob/1cc6ee1c711858b4f4af204d0044c62b70828ab8/extensions/xml/xml.language-configuration.json#L37 ?

Reply:
It's indeed a different pattern. But testing with node CLI gives the impression that those are equivalant (`'\u00C0' === '\u{C0}'` returns true), and this has the advantage of being parsed with the Java regexp as well as with the JS regexp.

Reply:
There are at least two different things between the XML element name pattern I used inlemminx and the one used in VSCode:
- The VSCode pattern is a full pattern describing a whole word (contains starting character definition and the following character definitions) - it's to be used when searching for a "word match", while in lemminx we're verifying the only last typed character (or a single character at some position), so the pattern is really a wordpart-like pattern (doesn't define a first word character).  I think it's still possible to make lemminx to search for the word by a linkedEditingRange region and use a whole word pattern (VSCode-like) to verify the word found, but imho it will add some possibility to fail for the validation (all this is a post processing, so trying to peek up a word by a linkedEditingRegion may be erroneous as the document may be change at  the moment of the event processing). However imho it's still possible.
- A minor difference - my pattern doesn't allow`\\u{10000}-\\u{EFFFF}` character codes, I'll modify the PR in order to enable these characters.